### PR TITLE
Support pytest6

### PR DESCRIPTION
**Pull request by tomdottom**
Addresses #18 

- Updates `BanditItem` to adhere to the new pytest6 api surface (uses `.from_parent` )
- Falls back to pytest5 api if `pytest.Item` is missing `.from_parent` class method
- Updates `tox.ini` to explicitly test pytest5 and pytest6 for each python version
- Updates `.travis.yaml` to explicitly test pytest5 and pytest6 for each python version

### CODE REVIEW

**Comment by tomdottom** [comment]
🤔 i May have to rebase

**Comment by tomdottom** [comment]
@mattwwarren the breakage on `py35-pytest6` disappears if the `--pycodestyle` option if removed from `pytest.ini`

Do you want to try and work around this (maybe a fancy `tox.ini` file). Or maybe just stop supporting py35?

**Comment by mattwwarren** [comment]
Since 3.5 was EOL in September, I think it's safe to say we can drop support

**Comment by tomdottom** [comment]
Python 3.5 support removed.  Test pass locally, but for some reason I don't see TravisCi running on this PR anymore?

File: .travis.yml [hunk 1]
```diff
@@ -6,24 +6,31 @@ language: python
 
 matrix:
     include:
-      - python: 3.5
-        env: TOX_ENV=py35
+        env: TOX_ENV=py35-pytest6
```
Comment (mattwwarren) [inline .travis.yml:9]: I think this is your travis failure problem. YAML syntax error

Comment (tomdottom) [reply .travis.yml]: Good spot.

**Comment by mattwwarren** [comment]
Looks like the tests passed now. Feeling good about this one?

**Pull request merged**